### ISSUE TITLE
Any plans to support Alpine Linux?

### ISSUE THREAD

**Issue by twoism-dev** [skylightio/skylight-ruby#59]
See also #35

**Comment by deepj** [comment]
Yes, it'd be very useful for more people here. Thank you for your time on spending on this support!

**Comment by wagenet** [comment]
We'll look into what it would take to support it. How soon we can do it depends on the required effort.

**Comment by loicboutet** [comment]
I would need the Alpine support as well. Any update?

**Comment by Micka33** [comment]
Hi guys !

Any news on Alpine-linux support ?
At Effilab, our ECS cluster runs fully on Alpine-linux.

**Comment by franzejr** [comment]
I would be thrilled with this support..Looking forward to see this too.

**Comment by LouisKottmann** [comment]
With Alpine being the recommended docker base image, as more apps get containerized, I believe this is mandatory for your business...

**Comment by deepj** [comment]
@LouisKottmann I agree!

**Comment by adilbenseddik** [comment]
mmh..
```CC=musl-gcc ./configure
prefix=/usr/local/musl```
http://wiki.musl-libc.org/wiki/Main_Page

**Comment by wagenet** [comment]
@adilbenseddik it's a bit more complex that that. However, we are working on it and hope to have something available soon.

**Comment by adilbenseddik** [comment]
@wagenet was kidding.. I bet you have more issues to address.. yet, we're really eager to see your app up and running!! 👍  (I'm I wrong if I say you struggle with upgrading your c++ code to musl libc ??)

**Comment by wagenet** [comment]
@adilbenseddik fortunately, we're using Rust and it actually has pretty good support for musl. I believe I've got everything working, but need a little bit more internal testing before we release.

**Comment by adilbenseddik** [comment]
hey!! @wagenet that sounds really great!! do you think we could have a pre-release aka. staging of your app? here at Effilab we are pretty serious about implementing your solution in our stack, so we may help you with ad hoc (client) feedback, and further (specific) tests results..  ??

**Comment by dlupu** [comment]
looking forward for the release. this is something we're clearlt missing also on our project.

**Comment by wagenet** [comment]
I just pushed out an initial stab at musl support to the master branch in this repo. If any of you would be up for testing it, that would be awesome. I did some basic tests on Alpine and it seems to work fine, but we don't use Alpine ourselves, so it's entirely possible that we missed some things.

**Comment by LouisKottmann** [comment]
Hello @wagenet, I'm the tester, on it.

**Comment by LouisKottmann** [comment]
So it does not work on my end, I can send you the puma log if you like (contains the core dump as well)

**Comment by tombruijn** [comment]
Hi Tom from @AppSignal here,

We're running into the same issue with our extension written in Rust. We tried to compile it with musl (the libc replacement Alpine linux uses), but no luck. Our agent runs fine, but the library we install in our Ruby gem as a native extension does not link properly.

We traced it back to a Rust issue with support for musl, reported here https://github.com/rust-lang/rust/issues/37007

Hope this helps!

**Comment by wagenet** [comment]
@LouisKottmann any information you can provide would be great.

**Comment by LouisKottmann** [comment]
@wagenet I've sent it by mail to you

**Comment by jlambert121** [comment]
Does https://github.com/rust-lang/rfcs/pull/1721 allow this to be resolved now?

**Comment by ryansch** [comment]
I just tried to install the skylight gem on a rails app inside of an alpine linux 3.4 container and I get a ruby segfault when I try to start rails.  I'm looking forward to progress in this area.  Back to debian for now.

**Comment by habitullence** [comment]
I've also just run into this.

**Comment by wagenet** [comment]
@ryansch @habitullence is your error the same as #91?

**Comment by wagenet** [comment]
This should work in current master. If you're having problems, please create a new issue.

**Comment by ababich** [comment]
I already confirmed in #91 that SEGFAULT is alpine-to-skylight compatibility issue and its fairly easy to reproduce

switching to debian based ruby image fixes the issue

**Comment by wagenet** [comment]
Others can follow the discussion in #91 but I can say for a fact that Skylight does work on at least _some_ Alpine setups. I suspect the specific issue in #91 is related to the Docker image itself. This is not an uncommon issue when using Docker.

**Comment by wagenet** [comment]
@habitullence with any luck, this fixes your issue: 138bb9ab67edfe2c5253ab78964fef3f11405030

**Comment by habitullence** [comment]
Thanks @wagenet 👍
